app/models/spawn_job.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+#
+# EventAction derived class using Single Table Inheritance
+#
+# Public interface:
+#   run!(user)
+#
+class SpawnJob < EventAction
+  def run!(user)
+    Job.new(job_type_id: job_type.id, worker: user.workers.assignment_order.first, status: "DISPATCHED")
```

Comment:
I think rails lets you do `job_type: job_type` instead of doing the id, ever so slightly cleaner here